internal/ast/compiler/anonymous_structs_to_named.go

```diff
@@ -0,0 +1,125 @@
+package compiler
+
+import (
+	"github.com/grafana/cog/internal/ast"
+	"github.com/grafana/cog/internal/tools"
+)
+
+var _ Pass = (*AnonymousStructsToNamed)(nil)
+
+// AnonymousStructsToNamed turns "anonymous structs" into a named object.
+//
+// Example:
+//
+//	```
+//	Panel struct {
+//		Options struct {
+//			Title string
+//		}
+//	}
+//	```
+//
+// Will become:
+//
+//	```
+//	Panel struct {
+//		Options PanelOptions
+//	}
+//
+//	PanelOptions struct {
+//		Title string
+//	}
+//	```
+type AnonymousStructsToNamed struct {
```

Comment:
I couldn't find a satisfying way of dealing with "anonymous structs" in Python so I worked around the problem: no more anonymous structs.